app/lib/shared/analyzer_client.dart

```diff
@@ -53,8 +54,28 @@ class AnalysisView {
   final AnalysisData _data;
   final Summary _summary;
 
-  AnalysisView(this._data) :
-    _summary = new Summary.fromJson(_data.analysisContent);
+  AnalysisView._(this._data, this._summary);
+
+  factory AnalysisView(AnalysisData data) {
+    Summary summary;
+    try {
+      if (data != null &&
+          data.analysisStatus != AnalysisStatus.aborted &&
+          data.analysisContent != null) {
+        summary = new Summary.fromJson(data.analysisContent);
+      }
+    } catch (e, st) {
+      // don't block on faulty serialization
+      _logger.warning(
+          'Unable to read analysis content for package: '
+          '${data.packageName} ${data.packageVersion}',
+          e,
+          st);
```

Comment:
There is a chance that a new `analyzer` instance with a new `pana` version will produce a format that we don't parse for whatever reason. I'm planning to introduce a `panaVersion` parameter in the `frontend`->`analyzer` request so that if the `frontend` is on older, it still gets the older version.